test/gtest/uct/ib/test_dc.cc

```diff
@@ -475,48 +475,52 @@ UCS_TEST_SKIP_COND_P(test_dc, stress_iface_ops,
 
 UCT_DC_INSTANTIATE_TEST_CASE(test_dc)
 
-static const char *uct_dc_tx_policy_names[] = {
-    [UCT_DC_TX_POLICY_DCS]           = "dcs",
-    [UCT_DC_TX_POLICY_DCS_QUOTA]     = "dcs_quota",
-    [UCT_DC_TX_POLICY_RAND]          = "rand",
-    [UCT_DC_TX_POLICY_HW_DCS]        = "hw_dcs",
-    [UCT_DC_TX_POLICY_LAST]          = NULL
-};
 
-static resource policies_variations[UCT_DC_TX_POLICY_LAST];
+extern "C" const char *uct_dc_tx_policy_names[];
 
 class test_dc_flow_control : public test_rc_flow_control {
 protected:
     virtual void init()
     {
-        const resource* rsc = GetParam();
-        modify_config("DC_MLX5_TX_POLICY", uct_dc_tx_policy_names[rsc->variant]);
+        std::string policy_name = get_tx_policy_name();
+        if (!policy_name.empty()) {
+            modify_config("DC_MLX5_TX_POLICY", policy_name);
+        }
         test_rc_flow_control::init();
     }
 
 public:
-    static std::vector<const resource*>
+    static std::vector<const struct resource*>
     enum_resources(const std::string &tl_name)
     {
+        static std::array<struct resource, UCT_DC_TX_POLICY_LAST> variations;
+        auto resources = uct_test::enum_resources(tl_name);
+
+        if (resources.empty()) {
+            return resources;
+        }
+
         ucs_assert(tl_name == "dc_mlx5");
-        std::vector<const resource*> resources = uct_test::enum_resources(
-                tl_name);
-        for (int policy = UCT_DC_TX_POLICY_DCS; policy < UCT_DC_TX_POLICY_LAST;
-             policy++) {
-            policies_variations[policy]         = *resources.at(0);
-            policies_variations[policy].variant = policy;
-            policies_variations[policy].variant_name =
-                    uct_dc_tx_policy_names[policy];
-            resources.push_back(&policies_variations[policy]);
+
+        for (int policy = 1; policy < UCT_DC_TX_POLICY_LAST; policy++) {
+            variations[policy]              = *resources.at(0);
```

Comment:
why only the 1st resource?
we can use static local vector for the variants. if it's not empty, we initialize it
like here: 
https://github.com/openucx/ucx/blob/1114be038ab8f2c91f349b1e6452e0f25cbeb58b/test/gtest/uct/test_sockaddr.cc#L103

Reply:
it can be done over every device but I thought it might add too many variants and add too much time to CI